src/query/storage/promremote/storage_test.go

```diff
@@ -248,6 +249,28 @@ func TestWriteBasedOnRetention(t *testing.T) {
 		assert.Contains(t, err.Error(), "test err")
 		assert.NotNil(t, promLongRetention2.GetLastWriteRequest())
 	})
+
+	t.Run("wrap non 5xx errors as invalid params error", func(t *testing.T) {
+		reset()
+		promLongRetention.SetError("test err", http.StatusForbidden)
+		err := sendWrite(storagemetadata.Attributes{
+			Resolution: 10 * time.Minute,
+			Retention:  8760 * time.Hour,
+		})
```

Comment:
nit: There are quite a few places in the tests where `storagemetadata.Attributes` is defined inline. I think this adds quite a bit of noise. Maybe assign a few `storagemetadata.Attributes` variations to to variables and use those instead? E.g. 

```suggestion
		promLongRetention.SetError("test err", http.StatusForbidden)
		err := sendWrite(longRetentionStorageMetadata)
```